aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -52,6 +51,32 @@ impl ExecutorTask for AptosExecutorTask {
         });
 
         let log_context = AdapterLogSchema::new(self.id, txn_idx as usize);
+
+        // We process direct write set payload here, to ensure that VM does not have to lift
+        // storage-level abstractions into more fine-grained types used by the VM.
+        if let Some(change_set) = txn.as_valid_direct_write_set_payload() {
+            let execution_result =
+                self.vm
+                    .execute_direct_write_set_payload(base_view, change_set, &log_context);
+            return match execution_result {
+                Ok(output) => {
+                    // Direct payload triggers reconfiguration, and subsequent transactions are skipped.
+                    // Note that here we read all state keys in the change set from the base state view,
+                    // which is fine: we do not use the results of these reads anyway. Changes made by
+                    // this transaction are applied directly and also do not break any other outputs,
```

Comment:
where does reconfiguration come into play? I guess that's for future transactions, but if there is no dependence on in-the-block state, why does it have to perform reads at all? If it makes only special reads, maybe we can enforce more invariants on behavior/reads during execution.